tests/foreman/api/test_contentview.py

```diff
@@ -1167,226 +1239,262 @@ def test_positive_promote_rh_custom_spin(self):
+        with global admin account:
 
-        @assert: User with admin role for content view can perform all
-        Variations above
+        1. create a user with all content views permissions
+        2. create lifecycle environment
+        3. create 2 content views (one to delete, the other to manage)
 
-        @caseautomation: notautomated
+        @setup: create a user with all content views permissions
+
+        @assert: The user can Read, Modify, Delete, Publish, Promote the
+        content views
 
         @CaseLevel: Integration
         """
-        # Note:
-        # Obviously all of this stuff should work with 'admin' user
-        # but these tests require creating a user with admin permissions
-        # for Content Views
-        # Dev note: none of this stuff is integrated with foreman rbac yet
-        # As such, all variations in here subject to change.
-        # Variations:
-        #  * Read, Modify, Delete, Promote Publish, Subscribe
+        user_login = gen_string('alpha')
+        user_password = gen_string('alphanumeric')
+        lce = entities.LifecycleEnvironment(organization=self.org).create()
+        # create a role with all content views permissions
+        role = entities.Role().create()
+        for res_type in ['Katello::ContentView', 'Katello::KTEnvironment']:
+            permission = entities.Permission(resource_type=res_type).search()
+            entities.Filter(
+                organization=[self.org],
+                permission=permission,
+                role=role
+            ).create()
+        # create a user and assign the above created role
+        entities.User(
+            organization=[self.org],
+            role=[role],
+            login=user_login,
+            password=user_password
+        ).create()
+        content_view = entities.ContentView(organization=self.org).create()
+        cfg = get_nailgun_config()
+        cfg.auth = (user_login, user_password)
+        # Check that we cannot create random entity due permission restriction
+        with self.assertRaises(HTTPError):
+            entities.Domain(cfg).create()
+        # Check Read functionality
+        content_view = entities.ContentView(cfg, id=content_view.id).read()
+        # Check Modify functionality
+        entities.ContentView(
+            cfg, id=content_view.id, name=gen_string('alpha')).update(['name'])
+        # Publish the content view.
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        # Promote the content view version.
+        promote(content_view.version[0], lce.id)
+        # Check Delete functionality
+        content_view = entities.ContentView(organization=self.org).create()
+        content_view = entities.ContentView(cfg, id=content_view.id).read()
+        content_view.delete()
+        with self.assertRaises(HTTPError):
+            entities.ContentView(cfg, id=content_view.id).read()
```

Comment:
nitpick: `content_view.read()` would work as well, no need to init another entity